great_expectations_provider/operators/great_expectations.py

```diff
@@ -44,121 +78,404 @@ class GreatExpectationsOperator(BaseOperator):
+        """
+        Determines whether a checkpoint exists or need to be built, then
+        runs the resulting checkpoint.
+        """
         self.log.info("Running validation with Great Expectations...")
+        self.conn = BaseHook.get_connection(self.conn_id) if self.conn_id else None
+        self.conn_type = self.conn.conn_type if self.conn else None
 
-        # Instantiate the Data Context
-        self.log.info("Ensuring data context is valid...")
+        self.log.info("Instantiating Data Context...")
+        if self.data_asset_name:
+            self.build_runtime_datasources()
         if self.data_context_root_dir:
-            self.data_context: BaseDataContext = ge.data_context.DataContext(
+            self.data_context = ge.data_context.DataContext(
                 context_root_dir=self.data_context_root_dir
             )
         else:
-            self.data_context: BaseDataContext = BaseDataContext(
-                project_config=self.data_context_config
-            )
+            self.data_context = BaseDataContext(project_config=self.data_context_config)
+        if self.datasource:
+            self.data_context.datasources[self.datasource.name] = self.datasource
 
-        # Instantiate the Checkpoint
+        self.log.info("Creating Checkpoint...")
         self.checkpoint: Checkpoint
         if self.checkpoint_name:
             self.checkpoint = self.data_context.get_checkpoint(
                 name=self.checkpoint_name
             )
+        elif self.checkpoint_config:
+            self.checkpoint = instantiate_class_from_config(
+                config=self.checkpoint_config,
+                runtime_environment={"data_context": self.data_context},
+                config_defaults={"module_name": "great_expectations.checkpoint"},
+            )
         else:
-            config_dict = self.checkpoint_config.to_json_dict()
-
-            # TODO: Ensure proper Checkpoint/SimpleCheckpoint instantiation in core so this can be removed
-            if config_dict.get("class_name") == "Checkpoint":
-                for i in ["site_names", "notify_with", "notify_on", "slack_webhook"]:
-                    config_dict.pop(i) if i in config_dict else None
-
+            self.checkpoint_name = (
+                f"{self.data_asset_name}.{self.expectation_suite_name}.chk"
+            )
+            self.checkpoint_config = self.build_default_checkpoint_config()
             self.checkpoint = instantiate_class_from_config(
-                config=config_dict,
+                config=self.checkpoint_config,
                 runtime_environment={"data_context": self.data_context},
-                config_defaults={
-                    "module_name": "great_expectations.checkpoint"},
+                config_defaults={"module_name": "great_expectations.checkpoint"},
             )
 
-        if self.checkpoint_kwargs:
-            result = self.checkpoint.run(**self.checkpoint_kwargs)
```

Comment:
I am seeing that the code that used `checkpoint_kwargs` was deleted here; am I missing the place where it is used (perhaps differently)?  Thanks.